eisbuk-admin/src/store/selectors/customers.ts

```diff
@@ -22,3 +23,41 @@ export const getCustomersList = (state: LocalStore): Customer[] => {
   const customersInStore = getCustomersRecord(state);
   return Object.values(customersInStore);
 };
+
+/**
+ * Get a list of all the customers whose birthdays are today
+ * (not an actual firestore query but rather synced local store entry)
+ * @param state Local Redux Store
+ * @returns list of customers grouped by birthday
+ */
+export const getCustomersWithBirthday = (
+  state: LocalStore
+): CustomerBirthday => {
+  const customersInStore = getCustomersRecord(state);
+
+  // find index of birthday thats >= todays date to slice at that and concat the rest at the end
+  const sortedCustomers = Object.values(customersInStore).sort((a, b) =>
+    DateTime.fromISO(a.birthday)
+      .set({ year: 2000 })
+      .toString()
+      .localeCompare(
+        DateTime.fromISO(b.birthday).set({ year: 2000 }).toString()
+      )
+  );
+  const index = sortedCustomers.findIndex(
+    (c) =>
+      DateTime.now().startOf("day").set({ year: 2000 }) <=
+      DateTime.fromISO(c.birthday).startOf("day").set({ year: 2000 })
+  );
+  const rearrangedCustomer = sortedCustomers
+    .slice(index === -1 ? 0 : index)
+    .concat(sortedCustomers.splice(0, index));
+
+  return rearrangedCustomer.reduce(
+    (r, v, i, a, k = DateTime.fromISO(v.birthday).toFormat("dd/MM")) => (
```

Comment:
ISO dates are `YYYY-mm-dd`. We don't need to parse them and format them to get rid of the first five charaters. `v.birthday.substring(5)` should be enough.
Also I'm not a fan of single letter variables, unless they are `i` and `j`, and they're integers representing the index of a loop, or [a few other cases](https://www.codereadability.com/i-n-k-single-letter-variable-names/).

Reply:
I agree here (even though I use them extensively in Go), and they will be compiled down to single letter vars on bundling (to reduce code as much as possible), but that reduction (bundle) isn't meant to be read by human, but rather just the JS interpreter.